connect/client_test.go

```diff
@@ -6,13 +6,52 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	cluster_api "github.com/pomerium/zero-sdk/cluster"
 	"github.com/pomerium/zero-sdk/connect"
 	"github.com/pomerium/zero-sdk/token"
 )
 
+func TestConfig(t *testing.T) {
+	t.Parallel()
+
+	for _, tc := range []struct {
+		endpoint      string
+		connectionURI string
+		requireTLS    bool
+		expectError   bool
+	}{
+		{"", "", false, true},
+		{"http://", "", false, true},
+		{"https://", "", true, true},
+		{"localhost:8721", "", false, true},
+		{"http://localhost:8721", "dns:localhost:8721", false, false},
+		{"https://localhost:8721", "dns:localhost:8721", true, false},
+		{"http://localhost:8721/", "dns:localhost:8721", false, false},
+		{"https://localhost:8721/", "dns:localhost:8721", true, false},
+		{"http://localhost:8721/path", "dns:localhost:8721", false, true},
+		{"https://localhost:8721/path", "dns:localhost:8721", true, true},
+		{"http://localhost", "dns:localhost:80", false, false},
+		{"https://localhost:443", "dns:localhost:443", true, false},
```

Comment:
Should this one be just "https://localhost" (testing that it will default to port 443)?